File: src/panorama/scripts/lib/env.ts
namespace invk {
  export namespace Env {
    export enum Name {
      Development = "development",
      Production = "production",
    }

    export class Env {
      static Name = Name;

      constructor(public name = Game.IsInToolsMode() ? Name.Development : Name.Production) {}

      get development() {
        return this.name === Name.Development;
      }

      get production() {
        return this.name === Name.Production;
      }
    }
  }
}

File: game/scripts/vscripts/invk/game_mode/env.lua
local val = require("invk.lang.value")

--- Environment
--- @enum invk.Env
local M = {
  DEVELOPMENT = "development",
  PRODUCTION = "production",
}

--- @return boolean
function M.is_dev_mode()
  return IsInToolsMode() or GameRules:IsDev() or val.non_nil(Convars:GetBool("developer"), false)
end

return M
